kahuna/public/js/components/gr-collections-panel/gr-collections-panel.js

```diff
@@ -53,13 +54,56 @@ grCollectionsPanel.controller('GrNodeCtrl', ['collections', function(collections
 
 }]);
 
+grCollectionsPanel.controller('GrAddToCollectionCtrl',
+    ['mediaApi', '$scope', function(mediaApi, $scope) {
+
+    const ctrl = this;
+
+    ctrl.saved = false;
+
+    ctrl.addImagesToCollection = imagesJson => {
+        const imageIds = imagesJson.map(imageJson => imageJson.id);
+        const promises = imageIds.map(id => mediaApi.find(id).then(image =>
+            image.perform('add-collection', {body: {data: ctrl.collectionPath}})));
+
+        return Promise.all(promises);
+    };
+}]);
+
+grCollectionsPanel.directive('grAddToCollection',
+        ['$timeout', 'vndMimeTypes',
+        function($timeout, vndMimeTypes) {
 
-grCollectionsPanel.directive('grAddToCollection', [function() {
     const dragClassName = 'node__info--drag-over';
     return {
-        link: function(scope, element) {
-            element.on('drop', ev => {
-                // TODO add the image(s) to collection
+        restrict: 'A',
+        controller: 'GrAddToCollectionCtrl',
+        controllerAs: 'ctrl',
+        bindToController: true,
+        scope: {
+            collectionPath: '=grAddToCollection'
+        },
+        link: function(scope, element, _, ctrl) {
+            element.on('drop', jqEv => {
+                const ev = jqEv.originalEvent;
+                const el = element[0];
```

Comment:
Note that you should be able to use jQlite helpers on `element` to add/remove CSS classes instead?